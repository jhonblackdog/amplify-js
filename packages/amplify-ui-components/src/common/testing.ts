export const pixelThreshold = 10;
