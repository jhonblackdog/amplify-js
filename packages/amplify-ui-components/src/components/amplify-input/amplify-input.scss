.input {
  display: block;
  width: 100%;
  padding: 16px;
  font-size: 14px;
  color: #152939;
  background-color: #fff;
  background-image: none;
  border: 1px solid var(--amplify-input-border-color);
  border-radius: 3px;
  box-sizing: border-box;
  margin-bottom: 10px;

  &:focus {
    outline: none;
    border-color: orange;
  }

  &:disabled {
    opacity: 0.5;
  }
}
