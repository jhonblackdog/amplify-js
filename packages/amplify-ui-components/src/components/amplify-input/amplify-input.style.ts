import { css } from 'emotion';

export const input = css`
  display: block;
  width: 100%;
  padding: 16px;
  font-size: 14px;
  color: #152939;
  background-color: #fff;
  background-image: none;
  border: 1px solid #c4c4c4;
  border-radius: 3px;
  box-sizing: border-box;
  margin-bottom: 10px;

  :focus {
    outline: none;
    border-color: orange;
  }
`;