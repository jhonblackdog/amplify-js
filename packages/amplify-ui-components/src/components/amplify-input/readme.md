# amplify-input



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute     | Description                                                           | Type                    | Default     |
| ------------- | ------------- | --------------------------------------------------------------------- | ----------------------- | ----------- |
| `description` | `description` | The text of the description.  Goes just below the label.              | `string`                | `undefined` |
| `fieldId`     | `field-id`    | The ID of the field.  Should match with its corresponding input's ID. | `string`                | `undefined` |
| `onInput`     | --            | The callback, called when the input is modified by the user.          | `(arg0: Event) => void` | `undefined` |
| `type`        | `type`        | The input type.  Can be any HTML input type.                          | `string`                | `"text"`    |


## Dependencies

### Used by

 - [amplify-field](../amplify-field)
 - [amplify-form-field](../amplify-form-field)

### Graph
```mermaid
graph TD;
  amplify-field --> amplify-input
  amplify-form-field --> amplify-input
  style amplify-input fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
