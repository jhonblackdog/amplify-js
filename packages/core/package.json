{
  "name": "@aws-amplify/core",
  "version": "1.0.26-unstable.1",
  "description": "Core category of aws-amplify",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/src/index.js",
  "typings": "./dist/esm/src/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "tslint 'src/**/*.ts' && jest --coverage",
    "build-with-test": "npm test && npm run build",
    "build:cjs": "tsc -p tsconfig.es5.json && rollup -c",
    "build:esm": "tsc -p tsconfig.es2015.json",
    "build": "npm run clean && npm run build:cjs && npm run build:esm",
    "clean": "rimraf tsc-out dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'"
  },
  "react-native": {
    "./lib/ClientDevice": "./lib/ClientDevice/reactnative.js",
    "./lib/RNComponents": "./lib/RNComponents/reactnative.js",
    "./lib/StorageHelper": "./lib/StorageHelper/reactnative.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "devDependencies": {
    "find": "^0.2.7",
    "prepend-file": "^1.3.1",
    "prettier": "^1.7.4",
    "typescript": "^3.2.2"
  },
  "dependencies": {
    "aws-sdk": "2.329.0",
    "url": "^0.11.0"
  },
  "jest": {
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>../../node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(tsx?|jsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "json",
      "jsx"
    ],
    "testEnvironment": "jsdom",
    "mapCoverage": true,
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    },
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ]
  }
}
