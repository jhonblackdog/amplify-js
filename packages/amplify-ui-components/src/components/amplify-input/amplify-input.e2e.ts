import { newE2EPage } from '@stencil/core/testing';
import { pixelThreshold } from '../../common/testing';

describe('amplify-input', () => {
  it('renders', async () => {
    const page = await newE2EPage();

    await page.setContent(
      `<amplify-input field-id="id" description="Description test" placeholder="Placeholder test"></amplify-input>`,
    );
    const screenshot = await page.compareScreenshot('Amplify Input', {fullPage: true});
    expect(screenshot).toMatchScreenshot({ allowableMismatchedPixels: pixelThreshold });

    const inputElement = await page.find('amplify-input');
    expect(inputElement).not.toBeNull();
  });

  it('fires an onInput event when the contents of the box are changed', async () => {
    const page = await newE2EPage();

    await page.setContent(`<amplify-input></amplify-input>`);

    const func = jest.fn();
    await page.exposeFunction('exposedfunc', func);

    await page.$eval('amplify-input', (inputElement: any) => {
      inputElement.onInput = this.exposedfunc;
      inputElement.label = 'adding a label so that the component rerenders';
    });
    await page.waitForChanges();

    const input = await page.find('input');
    await input.press('8');
    expect(func).toBeCalledTimes(1);
    expect(func.mock.calls[0][0].isTrusted).toBe(true);
    const value = await input.getProperty('value');
    expect(value).toBe('8');
  });

  it('can have a checkbox input', async () => {
    const page = await newE2EPage();

    await page.setContent(`<amplify-input type='checkbox'></amplify-input>`);
    await page.waitForChanges();

    const input = await page.find('input');
    expect(input).toEqualAttribute('type', 'checkbox');
  });

  it('can have a number input', async () => {
    const page = await newE2EPage();

    await page.setContent(`<amplify-input type='number'></amplify-input>`);
    await page.waitForChanges();

    const input = await page.find('input');
    expect(input).toEqualAttribute('type', 'number');
  });
});
