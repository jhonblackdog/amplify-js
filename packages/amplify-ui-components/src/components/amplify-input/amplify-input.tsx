import { Component, Prop, Host, h } from '@stencil/core';
import { TextFieldTypes } from '../../common/types/ui-types';

@Component({
  tag: 'amplify-input',
  styleUrl: 'amplify-input.scss',
})
export class AmplifyInput {
  /** The ID of the field.  Should match with its corresponding input's ID. */
  @Prop() fieldId: string;
  /** The text of the description.  Goes just below the label. */
  @Prop() description: string | null;
  /** The input type.  Can be any HTML input type. */
  @Prop() type?: TextFieldTypes = 'text';
  /** The callback, called when the input is modified by the user. */
  @Prop() handleInputChange?: (inputEvent: Event) => void;
  /** (Optional) The placeholder for the input element.  Using hints is recommended, but placeholders can also be useful to convey information to users. */
  @Prop() placeholder?: string = '';
  /** (Optional) String value for the name of the input. */
  @Prop() name?: string;
  /** The value of the content inside of the input field */
  @Prop() value: string;
  /** Attributes places on the input element: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes */
  @Prop() inputProps?: object;
  /** Will disable the input if set to true */
  @Prop() disabled?: boolean;

  render() {
    return (
      <Host class="input-host">
        <input
          id={this.fieldId}
          aria-describedby={this.fieldId && this.description ? `${this.fieldId}-description` : null}
          type={this.type}
          onChange={event => this.handleInputChange(event)}
          placeholder={this.placeholder}
          name={this.name}
          class="input"
          value={this.value}
          disabled={this.disabled}
          {...this.inputProps}
        />
      </Host>
    );
  }
}
