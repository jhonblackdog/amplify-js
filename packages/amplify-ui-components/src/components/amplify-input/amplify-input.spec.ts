import { newSpecPage } from '@stencil/core/testing';
import { AmplifyInput } from './amplify-input.tsx'

describe('amplify-input', () => {
  it('renders correct HTML', async () => {
    const page = await newSpecPage({
      components: [AmplifyInput],
      html: `<amplify-input></amplify-input>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
