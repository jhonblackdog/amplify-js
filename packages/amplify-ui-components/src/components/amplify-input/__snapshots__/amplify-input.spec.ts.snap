// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-input renders correct HTML 1`] = `
<amplify-input>
  <input class="css-1s61rh1" type="text">
</amplify-input>
`;
